linkerd/app/outbound/src/http/logical.rs

```diff
@@ -18,92 +24,127 @@ impl<N> Outbound<N> {
         NSvc::Future: Send,
     {
         self.map_stack(|config, rt, concrete| {
-            let config::ProxyConfig {
-                buffer_capacity,
-                cache_max_idle_age,
-                dispatch_timeout,
+            let crate::Config {
+                discovery_idle_timeout,
+                http_request_buffer,
                 ..
-            } = config.proxy;
+            } = config;
 
-            // Distribute requests over a distribution of balancers via a
-            // traffic split.
-            //
-            // If the traffic split is empty/unavailable, eagerly fail requests.
-            // When the split is in failfast, spawn the service in a background
-            // task so it becomes ready without new requests.
-            let logical = concrete
+            let split = concrete
                 .push_map_target(Concrete::from)
+                // This failfast is needed to ensure that a single unavailable
+                // balancer doesn't block the entire split.
+                //
+                // TODO(ver) remove this when we replace `split` with
+                // `distribute`.
+                .push_on_service(
+                    svc::layers()
+                        .push(svc::layer::mk(svc::SpawnReady::new))
+                        .push(svc::FailFast::layer(
+                            "HTTP Balancer",
+                            http_request_buffer.failfast_timeout,
+                        ))
+                )
                 .check_new_service::<(ConcreteAddr, Logical), _>()
+                // Distribute requests over a distribution of balancers via a
+                // traffic split.
                 .push(profiles::split::layer())
                 .push_on_service(
                     svc::layers()
-                        .push(svc::layer::mk(svc::SpawnReady::new))
                         .push(
                             rt.metrics
                                 .proxy
                                 .stack
                                 .layer(stack_labels("http", "logical")),
                         )
-                        .push(svc::FailFast::layer("HTTP Logical", dispatch_timeout))
-                        .push_spawn_buffer(buffer_capacity),
+                        .push_buffer("HTTP Logical", http_request_buffer),
                 )
-                .push_cache(cache_max_idle_age);
+                // TODO(ver) this should not be a generalized time-based evicting cache.
+                .push_cache(*discovery_idle_timeout);
 
             // If there's no route, use the logical service directly; otherwise
             // use the per-route stack.
-            logical
-                .clone()
+            let route = (split.clone())
```

Comment:
huh, what are the parens doing here? AFAICT just writing `split.clone()` would be equivalent?

Reply:
(this isn't a big deal; i was just kinda curious about it)

Reply:
yeah but it will be

```rust
let route = split
    .clone()
    ...
```

so this is just tricking rustfmt. probably not worth it.

Reply:
no worries, i was just wondering if it was load-bearing!